src/mo_def/syntax.ml

```diff
@@ -186,6 +186,7 @@ and exp' =
   | LabelE of id * typ * exp                   (* label *)
   | BreakE of id * exp                         (* break *)
   | RetE of exp                                (* return *)
+  | ResVarE                                    (* name of result *)
```

Comment:
If I remember correctly, Motoko allows to name values of result tuples (unary too?). I.e.
``` Motoko
func foo() : (first : Int, second : Char) { ... }
```
or such. I doubt however that these names are available in the AST. If they are maintained, one could refer to them from invariants, etc. and the `ResVarE` could be made redundant.

Reply:
Surprise! It appears and is named:
``` Scheme
$ rlwrap moc -dp
Motoko compiler (source 0.11.1-112-g1afaf35e3)
> func foo() : (first : Int) = (8);
(Prog
  (LetD
    (VarP foo)
    (FuncE
      ???
      Local
      foo
      (TupP)
      (NamedT first (PathT (IdH Int)))
      =
      (LitE (PreLit 8 Nat))
    )
  )
)
let foo : () -> Int = <func>
```

Reply:
It's not in scope in the function body though, so you couldn't mention it in an `assert`

Reply:
Right, IIRC we pondered the "final" syntax of Motoko-san to have postconditions written _before_ the function body (and after the return declaration), so it could be in scope there. Please ignore my comment and stay with the big kludge of `assert:*` that I came up with. I am not proud of it 🥹